### PR TITLE
Add split_pem filter

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Implements parts of #542.

##### ISSUE TYPE
- New Plugin Pull Request

##### COMPONENT NAME
split_pem

### CODE REVIEW

File: tests/integration/targets/filter_split_pem/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
```
Comment (briantist) [inline tests/integration/targets/filter_split_pem/tasks/main.yml:9]: total  nit and I don't care if it stays, but is this warning still useful? I don't really see it in anything new

File: tests/integration/targets/filter_split_pem/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    msg: "{{ [] | community.crypto.split_pem }}"
+  ignore_errors: true
+  register: output
+
+- name: Validate error
+  assert:
+    that:
+      - output is failed
+      - output.msg == "The community.crypto.split_pem input must be a text type, not <class 'list'>" or
+        output.msg == "The community.crypto.split_pem input must be a text type, not <type 'list'>"
```
Comment (briantist) [inline tests/integration/targets/filter_split_pem/tasks/main.yml:65]: I'm fine with the current thing here if you prefer it, change is just a suggestion

```suggestion
      - output.msg is search("^The community.crypto.split_pem input must be a text type, not <(?:class|type) 'list'>$")
```

**Review by briantist — APPROVED** [review]

File: tests/integration/targets/filter_split_pem/tasks/main.yml (hunk 1 above)
Comment (felixfontein) [reply tests/integration/targets/filter_split_pem/tasks/main.yml:9]: I don't know how useful this really is, but I think it's better to keep it :)

**Comment by felixfontein** [comment]
@briantist thanks for reviewing this!

**Pull request merged**